integration-tests/common-test-code/src/test/java/com/vaadin/flow/quarkus/it/AbstractCdiIT.java

```diff
@@ -41,6 +41,17 @@ protected String getText(String id) {
         return findElement(By.id(id)).getText();
     }
 
+    protected void waitForCount(int expectedCount, String counter) {
+        waitUntil(driver -> {
+            try {
+                assertCountEquals(expectedCount, counter);
+                return true;
```

Comment:
not sure if the return values are matters, so not sure when an exception happens, then it is false, then we do not return the type, because it is void... So basically we swallow the exception and pass the test in the end...

maybe 

```java
    protected boolean waitForCount(int expectedCount, String counter) {
       return waitUntil(driver -> {
            try {
                assertCountEquals(expectedCount, counter);
                return true;
                            } catch (Exception ex) {
                return false;
            }
        }, 1);
    }
    
    and then:
    ```
    assertTrue(waitForCount(expectedCount, DESTROY_COUNT));
    ```

What do you think? Maybe exceptions will never happen in that case it is safer and no swallowed false positive test happens.

Reply:
The return value is used by the waitUntil function. If, after a timeout, the lambda still return false (or null) waitUntil will throw an exception.
So in this case the assertion on return value is not useful for the test

Reply:
👍 ok, so we will have an exception in the case I was referring to, fair enough! thanks.